### PR TITLE
New template NovelFull

### PR DESCRIPTION
**Pull request by idMysteries**
TODO:

- [x] ajax detector
- [x] downloader
- [x] search

### CODE REVIEW

**Comment by idMysteries** [comment]
@dipu-bd aaahhhh I do not know how to implement the option with multiple links to the list of chapters. Right now I'm using the ajax_url parameter.

**Comment by idMysteries** [comment]
I'm asking not only for this reason, but also because the bot for creating a crawler will not work correctly

**Comment by idMysteries** [comment]
image
I have an idea... I will make the detector of this script as a temporary solution.

**Comment by idMysteries** [comment]
@dipu-bd search function not good... Hmmm...

**Comment by idMysteries** [comment]
@dipu-bd ready!

File: lncrawl/templates/novelfull.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        )
+
+        return soup.select("ul.list-chapter > li > a, select > option")
+
+    def make_chapter_list_url(self, soup: BeautifulSoup, nl_id: str) -> str:
+        url = f"{self.home_url}ajax/chapter-archive?novelId={nl_id}"
+
+        script = soup.find("script", text=re.compile(r"ajaxChapterOptionUrl\s+="))
+        if isinstance(script, Tag):
+            url = f"{self.home_url}ajax-chapter-option?novelId={nl_id}"
```
Comment (dipu-bd) [inline lncrawl/templates/novelfull.py:77]: did you forget to retrieve the url from script?

Comment (dipu-bd) [reply lncrawl/templates/novelfull.py:77]: i thought that the idea was to parse the ajax url from soup, no?

Comment (idMysteries) [reply lncrawl/templates/novelfull.py:77]: No. The script tag is missing (None) on other sites where ajax is different.
if script then ajax-chapter-option else ajax/...

Comment (idMysteries) [reply lncrawl/templates/novelfull.py:77]: image

Comment (dipu-bd) [reply lncrawl/templates/novelfull.py:77]: got it.

**Comment by dipu-bd** [comment]
i made some further changes. added cleaners to some of the crawlers. and replace single with pagination soup template

**Comment by idMysteries** [comment]
@dipu-bd why pagination soup template??

**Pull request merged**